### PR TITLE
fish shell support

### LINKED ISSUE #417: Support Fish shell, or don't try to install into Fish with ZSH syntax
Hi! My corporate ~spyware~ MDM software has forced this tool onto my system. This would be fine, except that you install shell support into Fish using ZSH syntax.

This is in my `~/.config/fish/config.fish`:
```
# Generated by Hermit; START; DO NOT EDIT.
HERMIT_ROOT_BIN="${HERMIT_ROOT_BIN:-"$HOME/bin/hermit"}"
eval "$(test -x $HERMIT_ROOT_BIN && $HERMIT_ROOT_BIN shell-hooks --print --zsh)"
# Generated by Hermit; END; DO NOT EDIT.
######
```

This causes Fish to produce this error:
```
~/.config/fish/config.fish (line 8): Unsupported use of '='. In fish, please use 'set HERMIT_ROOT_BIN "${HERMIT_ROOT_BIN:-"$HOME/bin/hermit"}"'.
HERMIT_ROOT_BIN="${HERMIT_ROOT_BIN:-"$HOME/bin/hermit"}"
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
from sourcing file ~/.config/fish/config.fish
source: Error while reading file '/Users/jw/.config/fish/config.fish'
```

I checked for `--fish`, but it doesn't exist:
```
~/bin/hermit shell-hooks --print --fish
Usage: hermit shell-hooks (install-hooks)

Manage Hermit auto-activation hooks of a shell.

Flags:
  -h, --help                 Show context-sensitive help.
      --version              Show version.
  -d, --debug                Enable debug logging.
  -t, --trace                Enable trace logging.
  -q, --quiet                Disable logging and progress UI, except fatal errors ($HERMIT_QUIET).
      --level=auto           Set minimum log level (auto,trace,debug,info,warn,error,fatal) ($HERMIT_LOG).
      --lock-timeout=30s     Timeout for waiting on the lock ($HERMIT_LOCK_TIMEOUT)
      --env=KEY=VALUE;...    Extra environment variables to apply to environments.

Command flags:
  --zsh     Update Zsh hooks.
  --bash    Update Bash hooks.

hermit: error: unknown flag --fish, did you mean one of "--zsh", "--bash"?
```

Please either:

* Add `--fish` output for shell hooks, or
* Do not install into Fish until shell hooks are supported for Fish.

### PR DESCRIPTION
**Pull request by mightyguava**
Adds native support for [fish shell](https://fishshell.com/).

Since fish isn't POSIX compliant, all scripts were translated to fish syntax (mostly by ChatGPT). There's now a new `bin/activate-hermit.fish` script that needs to be added to hermit repos to enable fish support, which hermit will add only if you run `hermit init` again.

I'm new to fish, so there might be some errors in the scripts. But both activate and deactivate seem to work fine in all the repos I tested in.

Closes #417.

### CODE REVIEW

File: bin/hermit [hunk 1]
```diff
@@ -1,4 +1,6 @@
 #!/bin/bash
+#
+# THIS FILE IS GENERATED; DO NOT MODIFY
```
Comment (mightyguava) [inline bin/hermit:3]: changes from running `hermit init`. Let me know if I should remove

Comment (alecthomas) [reply bin/hermit:3]: Nope this is fine!

File: bin/hermit [hunk 2]
```diff
@@ -15,14 +17,27 @@ if [ -z "${HERMIT_STATE_DIR}" ]; then
...
-export HERMIT_DIST_URL="${HERMIT_DIST_URL:-https://github.com/cashapp/hermit/releases/download/stable}"
+export HERMIT_DIST_URL="${HERMIT_DIST_URL:-https://github.com/cashapp/hermit/releases/download/canary}"
 HERMIT_CHANNEL="$(basename "${HERMIT_DIST_URL}")"
 export HERMIT_CHANNEL
 export HERMIT_EXE=${HERMIT_EXE:-${HERMIT_STATE_DIR}/pkg/hermit@${HERMIT_CHANNEL}/hermit}
 
 if [ ! -x "${HERMIT_EXE}" ]; then
   echo "Bootstrapping ${HERMIT_EXE} from ${HERMIT_DIST_URL}" 1>&2
-  curl -fsSL "${HERMIT_DIST_URL}/install.sh" | /bin/bash 1>&2
+  INSTALL_SCRIPT="$(mktemp)"
```
Comment (alecthomas) [inline bin/hermit:27]: It's _really_ not great to have too much shell in `bin/hermit`, because it slows down every invocation of `hermit`. I think this would be fine to pull out into a separate script though, perhaps `bin/download-hermit` or something? Also can we we make this a separate PR?

**Review by alecthomas — CHANGES_REQUESTED** [review]
This is awesome! 

Can you aalso dd fish to [this table](https://github.com/cashapp/hermit/blob/0418180e1a705f000aea58066a90feb771ac7b31/integration/integration_test.go#L35) to add Fish support to the integration tests?

File: bin/hermit (hunk 2 above)
Comment (mightyguava) [reply bin/hermit:27]: I didn’t edit this. This is from `hermit init`

Comment (alecthomas) [reply bin/hermit:27]: Ah man 🤦

Comment (alecthomas) [reply bin/hermit:27]: Weird that I don't see that in any of my recent environments...so strange.

**Comment by alecthomas** [comment]
> This is awesome!
> 
> Can you also add fish to [this table](https://github.com/cashapp/hermit/blob/0418180e1a705f000aea58066a90feb771ac7b31/integration/integration_test.go#L35) to add Fish support to the integration tests?

Actually this is going to be a bit more work than that, let's just get it in as-is.

**Review by alecthomas — APPROVED** [review]
